### ISSUE TITLE
Assertion on exit in Debug builds

### ISSUE THREAD

**Issue by gkoreman** [fastbuild/fastbuild#166]
When the application is exiting the following assertion frequently gets called. The application is attempting to connect to another worker as the application shuts down.

2>  -------- ASSERTION FAILED --------
2>  D:\fastbuild\Code\Core\Network\Network.cpp(144): Assert: NetworkStartupHelper::IsStarted()
2>  -----^^^ ASSERTION FAILED ^^^-----

**Comment by ffulin** [comment]
I have seen this as well. The assertion is harmless in this case, the connection on the other thread will fail gracefully. I will leave this issue open to track this problem though, as it is annoying in Debug builds.

**Comment by samhocevar** [comment]
The problem is caused by `Network::GetHostIPFromName()` relying on `NetworkStartupHelper::IsStarted()` to know whether a shutdown is in progress to allow for early exits. But this function itself requires network support (in the name resolution thread).

My suggestion:
   * create a `NetworkStartupHelper` in `Network::GetHostIPFromName()`
   * remove the `NetworkStartupHelper::IsStarted()` check
   * find another way to detect global shutdowns (optional in my opinion, since for now the timeout for name resolutions is 1 second, which is reasonable delay)

**Comment by ffulin** [comment]
In the end I decided to simply remove this assert, since the code is actually functioning correctly. (https://github.com/fastbuild/fastbuild/commit/f4b56d13c85e37a7f5749e924eeef5ded161ebaa) 

I chose not to add a NetworkStartupHelper in the GetHostIPFromName function, because as you noted, without an additional shutdown detection mechanism, that would increase the minimum build time by as much as a second: a non-trivial hit to incremental and no-op builds.

This fix will be in the source code for the next release (v0.94). I'll keep this issue open until that is available.
